### ISSUE TITLE
Perlmod does not generate grouped members from files

### ISSUE THREAD

**Issue by DGA45** [doxygen/doxygen#7494]
Some of our project were missing defines and enum in the generated Doxydocs.pm file, but those objects are correctly generated in the XML output.
When looking in the Doxygen code, we can see for the XML generator (xmlgen.cpp, function generateXMLForFile):
```
if (fd->getMemberGroupSDict())
  {
    MemberGroupSDict::Iterator mgli(*fd->getMemberGroupSDict());
    MemberGroup *mg;
    for (;(mg=mgli.current());++mgli)
    {
      generateXMLSection(fd,ti,t,mg->members(),"user-defined",mg->header(),
          mg->documentation());
    }
  }
```

But in the Perl generator, perlmodgen.cpp, function generatePerlModForFile, the management of getMemberGroupSDict is missing:
```
 m_output.closeList();
  
  generatePerlModSection(fd,fd->getMemberList(MemberListType_decDefineMembers),"defines");
  generatePerlModSection(fd,fd->getMemberList(MemberListType_decProtoMembers),"prototypes");
  generatePerlModSection(fd,fd->getMemberList(MemberListType_decTypedefMembers),"typedefs");
  generatePerlModSection(fd,fd->getMemberList(MemberListType_decEnumMembers),"enums");
  generatePerlModSection(fd,fd->getMemberList(MemberListType_decFuncMembers),"functions");
  generatePerlModSection(fd,fd->getMemberList(MemberListType_decVarMembers),"variables");

  addPerlModDocBlock(m_output,"brief",fd->getDefFileName(),fd->getDefLine(),0,0,fd->briefDescription());

```

Additionally, the XML generator is using a more portable management of the memberlists, using the following code:
```
  QListIterator<MemberList> mli(fd->getMemberLists());
  MemberList *ml;
  for (mli.toFirst();(ml=mli.current());++mli)
  {
    if ((ml->listType()&MemberListType_declarationLists)!=0)
    {
      generateXMLSection(fd,ti,t,ml,g_xmlSectionMapper.find(ml->listType()));
    }
  }
```
The same mechanism should be used for the PerlMod generator, instead of multiples lines with "hard-coded" list types.

**Comment by albert-github** [comment]
perlmod is not used often / hardly and there have been ideas to remove it.

@doxygen what is the status?

@DGA45 

For what purpose do you use the perlmod output (just to get a feeling).

-   Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.
-    Please also specify the doxygen version used.

**Comment by DGA45** [comment]
I described our purpose in the #7490 
Here is an example to reproduce the issue:
[Doxy issues example.zip](https://github.com/doxygen/doxygen/files/4035994/Doxy.issues.example.zip)

**Comment by albert-github** [comment]
Code of pull request #7565  has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.18.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
